### PR TITLE
Avoid unnecessary intermediate String.IndexOf calls

### PR DESCRIPTION
**Pull request by justinvp**

### CODE REVIEW

File: src/mscorlib/src/System/String.Searching.cs [hunk 1]
```diff
@@ -113,15 +113,6 @@ public partial class String
...
         [Pure]
         public int IndexOf(String value, int startIndex, int count) {
-            if (startIndex < 0 || startIndex > this.Length) {
-                throw new ArgumentOutOfRangeException(nameof(startIndex), Environment.GetResourceString("ArgumentOutOfRange_Index"));
-            }
-
-            if (count < 0 || count > this.Length - startIndex) {
-                throw new ArgumentOutOfRangeException(nameof(count), Environment.GetResourceString("ArgumentOutOfRange_Count"));
-            }
-            Contract.EndContractBlock();
```
Comment (justinvp) [inline src/mscorlib/src/System/String.Searching.cs]: Note that removing the validation here does introduce a subtle behavior difference: if `value` is null, `ArgumentNullException` will be thrown now before the other arguments are validated. I'm guessing this is an acceptable change, but let me know if not and I'll revert this part of the change.

**Comment by justinvp** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test please

File: src/mscorlib/src/System/String.Searching.cs (hunk 1 above)
Comment (danmosemsft) [inline src/mscorlib/src/System/String.Searching.cs]: I would rather revert that, as this is such a heavily used type

Comment (justinvp) [inline src/mscorlib/src/System/String.Searching.cs]: Reverted

**Comment by danmosemsft** [comment]
LGTM

**Comment by stephentoub** [comment]
What's the value of this change?

**Comment by justinvp** [comment]
> What's the value of this change? Are the intermediates not getting inlined? We're duplicating some code, so I'm wondering what it buys us.

I haven't checked the inlining yet... The reason I submitted this change was because I was looking at cleaning up the formatting of this file and noticed all the other overloads in this file are all calling the largest overload directly instead of chaining. This change makes these three methods consistent with the rest of the file in that regard. If there's concern over some duplicated code, should we change the rest of the file to call the next largest overload (chain), instead of directly calling the largest overload?

**Comment by justinvp** [comment]
Going to close this as there doesn't seem to be a clear benefit.

**Pull request closed without merging**